### PR TITLE
:building_construction:  Refactor code as SOLID :hammer_and_wrench: 

### PR DESCRIPTION
**Pull request by SimonGideon**
## This PR includes the following:

- Describe the SOLID principles of OOP.
- Refactor your own Ruby code after gaining new knowledge.
- Apply Ruby best practices and language style guides in code.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -36,17 +40,24 @@ def handle_option(option)
...
     when 6
-      handle_list_rentals
+      @rentals_handler.handle_list_rentals
     else
       puts 'Invalid option'
     end
   end
+end
+
+# person handler
```
Comment (NduatiKagiri) [inline main.rb]: - [ ] According to the project requirements, this should be the end of the `main.rb` file. Kindly the rest of the code (PersonHandler, Rentals, and BooksaHandler classes) into a seperate `app.rb` file and import it to main. Please also remember to check that your app still works as expected after making the changes.

**Review by NduatiKagiri — CHANGES_REQUESTED** [review]
Hi @SimonGideon 👋 ,

# CHANGES REQUIRED ♻️  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights ✔️ 
✅ GitFlow observed
✅ No Linters errors
✅ Good PR description

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (SimonGideon) [reply main.rb]: @NduatiKagiri Thanks for the review the requested changes will be implemented. :slightly_smiling_face:

**Review by mukhammadrizooff — APPROVED** [review]
Hi @SimonGideon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**